lib/puppet/parser/scope.rb

```diff
@@ -652,8 +652,10 @@ def has_enclosing_scope?
   private :has_enclosing_scope?
 
   def qualified_scope(classname)
-    raise _("class %{classname} could not be found") % { classname: classname }     unless klass = find_hostclass(classname)
-    raise _("class %{classname} has not been evaluated") % { classname: classname } unless kscope = class_scope(klass)
+    klass = find_hostclass(classname)
+    kscope = class_scope(klass)
+    raise _("class %{classname} could not be found") % { classname: classname }     unless klass
```

Comment:
I think we need to check that `klass` isn't nil before calling `class_scope(klass)`